runtime/System/Console/Hawk/Args/Spec.hs

```diff
@@ -35,8 +35,10 @@ data InputSource
 
 data OutputSink
     = UseStdout
-    -- OutputFile FilePath  -- we might want to implement --in-place
-                            -- in the future
+    | OutputFile
+        { outputFile :: String
+        , backupFile :: Maybe String
+        }
```

Comment:
Please don't mix record syntax and sum types, this causes `outputFile` and `backupFile` to be become partial functions:

    > outputFile UseStdout
    Exception: No match in record selector outputFile

Reply:
I would create a separate record type, like this:

    data FileSink = FileSink
      { outputFile :: FilePath
      , backupFile :: Maybe FilePath
      }
    
    data OutputSink
      = UseStdout
      | OutputFile FileSink